extensions/spring-web/runtime/src/main/java/io/quarkus/spring/web/runtime/ResponseStatusExceptionMapper.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.spring.web.runtime;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+import org.jboss.resteasy.specimpl.ResponseBuilderImpl;
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.server.ResponseStatusException;
+
+public class ResponseStatusExceptionMapper implements ExceptionMapper<ResponseStatusException> {
+
+    @Override
+    public Response toResponse(ResponseStatusException exception) {
+        Response.ResponseBuilder responseBuilder = new ResponseBuilderImpl().status(exception.getStatus().value());
+        addHeaders(responseBuilder, exception.getResponseHeaders());
+        return responseBuilder.entity(exception.getMessage())
+                .type(MediaType.TEXT_PLAIN).build();
```

Comment:
Does Spring always send `text/plain`? I would expect it to be clever than that.

Reply:
Fair point, but we have been using `text/plain` for exceptions annotated with `@ResponseStatus` (which is very similar to this) and we haven't had anyone complain so far.

We can certainly reexamine this on a grander scale if it becomes a problem.